app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -806,5 +806,25 @@ def render_podman_error(code, message, status = :bad_request)
     def item_not_found(item)
       render_podman_error("NAME_UNKNOWN", _("%s was not found!") % item, :not_found)
     end
+
+    def static_index
+      host_ip = request.remote_ip
+      host = ::Host.joins(:primary_interface).where(nics: { ip: host_ip }).first
+      flatpak_index = (redirect_client { Resources::Registry::Proxy.get(@_request.fullpath, headers) })
+      flatpak_index_json = JSON.parse(flatpak_index)
+      # Filter out repositories if it's a registered host
+      if host&.content_view_environments&.any?
```

Comment:
If host is assigned only to Library, this codepath would still run. Maybe flip the conditional and say `if host.default_environment?` return the unfiltered index, otherwise do all this stuff. thoughts?

Reply:
We still want to scope by the organization of the Library environment..